app/lib/admin/backend.dart

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:client_data/admin_api.dart' as api;
+import 'package:continuation_token/continuation_token.dart';
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+
+import '../account/backend.dart';
+import '../account/models.dart';
+import '../package/models.dart';
+import '../publisher/models.dart';
+import '../shared/configuration.dart';
+import '../shared/exceptions.dart';
+
+final _logger = Logger('pub.admin.backend');
+
+/// Sets the admin backend service.
+void registerAdminBackend(AdminBackend backend) =>
+    ss.register(#_adminBackend, backend);
+
+/// The active admin backend service.
+AdminBackend get adminBackend => ss.lookup(#_adminBackend) as AdminBackend;
+
+/// Represents the backend for the admin handling and authentication.
+class AdminBackend {
+  final DatastoreDB _db;
+  AdminBackend(this._db);
+
+  Future<R> _withAdmin<R>(Future<R> fn(AuthenticatedUser user)) async {
+    return await withAuthenticatedUser((u) async {
+      // TODO: remove once [withAuthenticatedUser] is using [User]
+      final user = await accountBackend.lookupUserById(u.userId);
+      if (!activeConfiguration.admins.contains(user.oauthUserId)) {
+        throw AuthorizationException.userIsNotAdminForPubSite();
+      }
+      return await fn(u);
+    });
+  }
+
+  /// List users.
+  Future<api.AdminListUsersResponse> listUsers(
+      {String continuationToken, int limit = 1000}) async {
+    return await _withAdmin((user) async {
+      final query = _db.query<User>()
+        ..order('__key__')
+        ..limit(limit);
+
+      if (continuationToken != null) {
+        Map<String, dynamic> map;
+        try {
+          map = _decodeContinuation(user.userId, continuationToken);
+        } on FormatException catch (_) {
+          throw InvalidInputException.continuationParseError();
+        }
+        InvalidInputException.checkNotNull(map, 'continuationToken');
+
+        final id = map['id'] as String;
+        // NOTE: we should fix https://github.com/dart-lang/gcloud/issues/23
+        //       and remove the toDatastoreKey conversion here.
+        final key =
+            _db.modelDB.toDatastoreKey(_db.emptyKey.append(User, id: id));
+        if (id != null) {
+          query.filter('__key__ >', key);
+        }
+      }
```

Comment:
I think this continuation token logic is too complex. Let's keep it simple:
```dart
import 'package:convert' show hex;

String _decodeContinuationToken(String continuationToken) {
  try {
    return hex.fuse(utf8).decode(continuationToken);
  } on FormatException {
    throw InvalidInputException.invalidContinuationToken();
  }
}

....
```

That'll leave us with:
```dart
// Add key to start from, if continuation token is given.
if (continuationToken != null) {
  query.filter('__key__ >', _db.modelDB.toDatastoreKey(_db.emptyKey.append(
    User,
    id: _decodeContinuationToken(continuationToken),
  ));
}
```

Reply:
I'll rewrite `package:continuation_token` to support the encoding of single Strings. I'd like to keep the base64 encoding as it is shorter than hex.